### PR TITLE
fix: [CDS-58542]: remove userInvite from the list of usergroups

### PR DESCRIPTION
**Pull request by rafaelmart1ns**
## Describe your changes
The query made to list a user's usergroups via graphlq uses data from a usergroups invite table that contains data from the usergroups to which the user was assigned when creating the account.
And if that user is removed, that table still continues with the data and displaying the data of that usergroup that the user no longer belongs to.
In this pr, the use of data from this table to list data about usergroups was removed.

Video explained with changes and without.

https://user-images.githubusercontent.com/108759997/230940945-7f5364a4-0d27-4e8c-9cac-d4a491cae9d8.mov

https://user-images.githubusercontent.com/108759997/230941013-6373d712-7b6b-41e6-9ce2-dd999cfafac9.mov

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/46194)

### CODE REVIEW

**Review by lasdox — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46194)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by lasdox — DISMISSED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46194)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by rafaelmart1ns — COMMENTED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46194)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by vcalasansh — DISMISSED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46194)* status: :shipit: complete! ✓

**Comment by AnshuH** [comment]
trigger review

**Comment by rafaelmart1ns** [comment]
trigger review

File: 400-rest/src/main/java/software/wings/graphql/datafetcher/userGroup/UserGroupQueryHelper.java [hunk 1]
```diff
@@ -71,29 +65,10 @@ public void setQuery(List<QLUserGroupFilter> filters, Query query) {
...
-    userInviteField.in(emailIds);
-    Set<String> userGroups = new HashSet<>();
-    try (HIterator<UserInvite> iterator = new HIterator<>(userInviteQuery.fetch())) {
-      while (iterator.hasNext()) {
-        userGroups.addAll(iterator.next().getUserGroups().stream().map(UserGroup::getUuid).collect(Collectors.toSet()));
-      }
-    }
-
-    Query<UserGroup> userGroupsQuery = populateAccountFilter(UserGroup.class);
-    CriteriaContainer userGroupsCriteria = userGroupsQuery.criteria("_id").in(userGroups);
```
Comment (meenakshi-harness) [inline 400-rest/src/main/java/software/wings/graphql/datafetcher/userGroup/UserGroupQueryHelper.java:92]: Why are we removing this whole flow. Can we sync on this?

File: 400-rest/src/main/java/software/wings/graphql/datafetcher/userGroup/UserGroupQueryHelper.java [hunk 2]
```diff
@@ -71,29 +65,10 @@ public void setQuery(List<QLUserGroupFilter> filters, Query query) {
...
     if (memberIdsCriteria == null) {
       return;
     }
-    query.or(memberIdsCriteria, userGroupsCriteria);
```
Comment (meenakshi-harness) [inline 400-rest/src/main/java/software/wings/graphql/datafetcher/userGroup/UserGroupQueryHelper.java:96]: Can you add the testing done for this 
1. What is expected response.
2. What is existing response and new response ( Add screenshot preferably)
3. UTs

**Review by meenakshi-harness — APPROVED** [review]

**Review by meenakshi-harness — CHANGES_REQUESTED** [review]
Can you add the testing done for this

What is expected response.
What is existing response and new response ( Add screenshot preferably)
UTs
Test this change for new users as well

File: 400-rest/src/main/java/software/wings/graphql/datafetcher/userGroup/UserGroupQueryHelper.java (hunk 1 above)
Comment (rafaelmart1ns) [reply 400-rest/src/main/java/software/wings/graphql/datafetcher/userGroup/UserGroupQueryHelper.java:92]: Its synced up.

File: 400-rest/src/main/java/software/wings/graphql/datafetcher/userGroup/UserGroupQueryHelper.java (hunk 2 above)
Comment (rafaelmart1ns) [reply 400-rest/src/main/java/software/wings/graphql/datafetcher/userGroup/UserGroupQueryHelper.java:96]: I added a video in the description as an example of behaviour.
It added UT.

**Review by vcalasansh — APPROVED** [review]
> Reviewed 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46194)* status: 2 file(s) to review. 2. unresolved discussions.

**Comment by rafaelmart1ns** [comment]
trigger review

**Pull request merged**